prow/pod-utils/gcs/upload.go

```diff
@@ -83,11 +91,27 @@ func upload(dtw destToWriter, uploadTargets map[string]UploadFunc) error {
 		log := logrus.WithField("dest", dest)
 		log.Info("Queued for upload")
 		go func(f UploadFunc, writer dataWriter, log *logrus.Entry) {
-			sem.Acquire(context.Background(), 1)
-			defer sem.Release(1)
 			defer group.Done()
-			if err := f(writer); err != nil {
+
+			var err error
+
+			for retryIndex := 1; retryIndex <= retryCount; retryIndex++ {
+				sem.Acquire(context.Background(), 1)
+				log.Debugf("Attempting upload %v", retryIndex)
```

Comment:
Can we get this log with a `retryIndex > 1` and say something like `log.WithField("retry_attempt", retryIndex).Debug("Retrying upload")`?